lsp-mode.el

```diff
@@ -3474,46 +3388,45 @@ in that particular folder."
       (lsp--remove-overlays 'lsp-links)
 
       (remove-hook 'xref-backend-functions #'lsp--xref-backend t)
-      (setq-local global-mode-string (remove status global-mode-string))
-      (when (bound-and-true-p company-mode)
-        (lsp--clean-company))))))
+      (setq-local global-mode-string (remove status global-mode-string))))))
 
 (defun lsp-configure-buffer ()
-  (when lsp-auto-configure
```

Comment:
why do we remove lsp-auto-configure check?

Reply:
I think that we should probably remove the dual check of `lsp-auto-enable` and `lsp-xyz-enable` everywhere.
Instead we can set the `lsp-xyz-enable` to `nil` for feature that need auto enable to be used, and `t` otherwise.

The `lsp-auto-enable` will only being used in `lsp--auto-configure` and will set all the `lsp-xyz-enable` to t. Or even better, we can expose `lsp--auto-configure-hook` (internal hook) so each feature can add its own auto configure.

Reply:
you mean then that `lsp-auto-enable` will always override user definition of `lsp-xyz-enable` setting it to `t`?

Reply:
> The `lsp-auto-enable` will only being used in `lsp--auto-configure` and will set all the `lsp-xyz-enable` to t. Or even better, we can expose `lsp--auto-configure-hook` (internal hook) so each feature can add its own auto configure.

So, if someone wants to turn off one feature he should turn off auto-enable and then go and enable the rest one? Is that what you are suggesting?

Reply:
Yeah, this sounds bad to me, I think we should keep the `lsp-auto-enable` just enabling the features if `lsp-xyz-enable` is `t`

Reply:
We can look at that from different angle, until now if we want to enable just an optional feature like `lens`, we have to enable `lsp-auto-enable` and disable the rest. Which is not natural at all.

If we really want to archive that:
- User can easily turn off auto configure and enable one feature
- User can easily leave auto configure on and disable one feature
We should have new flags like `lsp-auto-enable-for-xyz`, which will really be over-killed.

How about we do some thing like checking `(get 'lsp-xyz-enable 'not-autoconfig)` in auto configure function and set `lsp-xyz-enable` to `t` if that property is not set?
With that we can still promote `lsp-xyz-enable` feature when auto config is on. And user can disable feature with `(set 'lsp-xyz-enable 'not-autoconfig)`